### ISSUE TITLE
Group update works depending on names of updated dependencies

### ISSUE THREAD

**Issue by scheleaap** [scala-steward-org/scala-steward#2716]
At our company, we release two libraries together with the same version number:
`com.example:foo-lastmile` and
`com.example:foo-scorer`.

Scala Steward does not update one project that relies on both dependencies. Through debugging, I was able to figure out that the names of the dependencies plays a role. Scala Steward [sorts updates by name](https://github.com/scala-steward-org/scala-steward/blob/fa6f7f0505e964b599a32c023cc4fe491c2f613b/modules/core/src/main/scala/org/scalasteward/core/update/PruningAlg.scala#L45); therefore, the `Update.Group` will contain `foo-lastmile` first and `foo-scorer` second.

This update fails, as the following test proves:
```scala
  test("fails") {
    val original =
      """private val version = "1.0.0"
        |
        |lazy val foo = "com.example" % "foo-lastmile" % version
        |lazy val foo = "com.example" % "foo-scorer" % version
        |""".stripMargin
    val expected =
      """private val version = "1.1.0"
        |
        |lazy val foo = "com.example" % "foo-lastmile" % version
        |lazy val foo = "com.example" % "foo-scorer" % version
        |""".stripMargin

    val update = ("com.example".g % Nel.of(
      "foo-lastmile".a,
      "foo-scorer".a,
    ) % "1.0.0" %> "1.1.0").group
    assertEquals(update.replaceVersionIn(original)._1, Some(expected))
  }
```

However, if we change the name of `foo-lastmile` to `foo-zzz` so that it comes _after_ `foo-scorer`, the test suddenly passes:
```scala
  test("passes") {
    val original =
      """private val version = "1.0.0"
        |
        |lazy val foo = "com.example" % "foo-zzz" % version
        |lazy val foo = "com.example" % "foo-scorer" % version
        |""".stripMargin
    val expected =
      """private val version = "1.1.0"
        |
        |lazy val foo = "com.example" % "foo-zzz" % version
        |lazy val foo = "com.example" % "foo-scorer" % version
        |""".stripMargin

    val update = ("com.example".g % Nel.of(
      "foo-scorer".a,
      "foo-zzz".a,
    ) % "1.0.0" %> "1.1.0").group
    assertEquals(update.replaceVersionIn(original)._1, Some(expected))
  }
```

In other words, the alphabetical order of the dependencies' names being updated plays a role in whether Scala Steward can update them or not, and I think this should not be the case.

This was tested on version 0.15.0.

I'd be happy to assist in investigating the problem. I tried to understand the update heuristics' code, but found the logic quite hard to follow, so I would need to pair on that with someone.

**Comment by terryhendrix1990** [comment]
I think this will solve your issue (as it does mine)
https://github.com/scala-steward-org/scala-steward/pull/2714
Hoping it will be done & approved soon

**Comment by exoego** [comment]
Heuristics to find version strings is a stiff part.
If we loose it, some unexpected files might be updated.
If we tighten it, some might not be updated.

To extract common version string into `val`, it is better to prefix the `val` with artifactIid, like `fooVersion`.
With `fooVersion`, the test passes.
 
```scala
    val original =
      """private val fooVersion = "1.0.0"
        |
        |lazy val foo = "com.example" % "foo-lastmile" % fooVersion
        |lazy val foo = "com.example" % "foo-scorer" % fooVersion
        |""".stripMargin
    val expected =
      """private val fooVersion = "1.1.0"
        |
        |lazy val foo = "com.example" % "foo-lastmile" % fooVersion
        |lazy val foo = "com.example" % "foo-scorer" % fooVersion
        |""".stripMargin

    val update = ("com.example".g % Nel.of(
      "foo-lastmile".a,
      "foo-scorer".a,
    ) % "1.0.0" %> "1.1.0").group
    assertEquals(update.replaceVersionIn(original)._1, Some(expected))
```

**Comment by scheleaap** [comment]
Thank you for taking a look, @exoego. You're right, renaming `version` to `fooVersion` works. I think this is a viable workaround for us, for now. It remains a delicate business though ;-).
